T/TracyProfiler/build_tarballs.jl

```diff
@@ -2,69 +2,167 @@
+    export LDFLAGS="-ldl ${LDFLAGS}"
 fi
 
-if [[ "${target}" == x86_64-apple-darwin* ]]; then
-    echo "Installing newer MacOS 11.0 SDK"
-
-    pushd $WORKSPACE/srcdir/MacOSX11.*.sdk
+# Install macOS 14.0 SDK for C++20 <ranges> support (needed by ppqsort)
+if [[ "${target}" == *-apple-darwin* ]]; then
+    echo "Extracting MacOSX14.0.sdk.tar.xz (this may take a while)"
     rm -rf /opt/${target}/${target}/sys-root/System
     rm -rf /opt/${target}/${target}/sys-root/usr/include/libxml2/libxml
-    cp -ra usr/* "/opt/${target}/${target}/sys-root/usr/."
-    cp -ra System "/opt/${target}/${target}/sys-root/."
-    popd
+    tar --extract \
+        --file=${WORKSPACE}/srcdir/MacOSX14.0.sdk.tar.xz \
+        --directory="/opt/${target}/${target}/sys-root/." \
+        --strip-components=1 \
+        --warning=no-unknown-keyword \
+        MacOSX14.0.sdk/System \
+        MacOSX14.0.sdk/usr
 fi
 
-atomic_patch -p1 ../patches/TracyProfiler-nfd-extended-1.0.2.patch
-atomic_patch -p1 ../patches/TracyProfiler-filter-user-text.patch
-atomic_patch -p1 ../patches/TracyProfiler-no-divide-zero.patch
-atomic_patch -p1 ../patches/TracyProfiler-rr-nopl-seq.patch
-
-# Build / install the profiler GUI
-make -e -j${nproc} -C profiler/build/unix LEGACY=1 IMAGE=tracy release
-cp -v ./profiler/build/unix/tracy* $bindir
-
-# Build / install the update utility
-make -e -j${nproc} -C update/build/unix IMAGE=tracy-update release
-cp -v ./update/build/unix/tracy* $bindir
-
-# Build / install the capture utility
-make -e -j${nproc} -C capture/build/unix IMAGE=tracy-capture release
-cp -v ./capture/build/unix/tracy* $bindir
-
-# Build / install the csvexport utility
-make -e -j${nproc} -C csvexport/build/unix IMAGE=tracy-csvexport release
-cp -v ./csvexport/build/unix/tracy* $bindir
-
-# Build / install the import-chrome utility
-make -e -j${nproc} -C import-chrome/build/unix IMAGE=tracy-import-chrome release
-cp -v ./import-chrome/build/unix/tracy* $bindir
+# Apply patch to skip ExternalProject for embed if pre-built binary exists
+atomic_patch -p1 ../patches/cross-compile-embed.patch
+
+# Apply patch to fix MSVC-specific /MP flag on MinGW
+atomic_patch -p1 ../patches/mingw-no-msvc-flags.patch
+
+# Apply patch to fix TracyPopcnt.hpp for MinGW (use GCC builtins instead of MSVC intrinsics)
+atomic_patch -p1 ../patches/mingw-popcnt.patch
+
+# Apply patch to fix 32-bit MinGW build (IMAGEHLP_LINE -> IMAGEHLP_LINE64 for SymGetLineFromAddr64)
+atomic_patch -p1 ../patches/mingw-imagehlp-line64.patch
+
+# Pre-build the 'embed' helper tool with the host compiler.
+# This tool embeds fonts/manual/prompts into C++ source during the build.
+# CMake's ExternalProject would build it for the target architecture, which fails
+# during cross-compilation (can't run aarch64 binary on x86_64 host).
+# The patch we applied makes CMake skip ExternalProject if embed already exists.
+echo "Building embed helper for host..."
+mkdir -p build/profiler
+${CXX_BUILD:-c++} -std=c++20 -O2 \
```

Comment:
`c++` would be wrong, that's the target compiler
```suggestion
${CXX_BUILD} -std=c++20 -O2 \
```

Reply:
Got it, thanks!